tdrs-backend/tdpservice/search_indexes/utils.py

```diff
@@ -0,0 +1,284 @@
+        paginator = Paginator(qset, settings.BULK_CREATE_BATCH_SIZE)
+        for page in paginator:
+            try:
+                doc().update(page.object_list, refresh=True, action="delete")
+            except ElasticsearchException:
+                elastic_exceptions[model] = elastic_exceptions.get(model, 0) + 1
+                continue
+
+
+def delete_records(file_ids, new_indices, log_context):
+    """Delete records, errors, and documents from Postgres and Elastic."""
+    total_deleted = 0
+    elastic_exceptions = dict()
+    for doc in DOCUMENTS:
+        try:
+            model = doc.Django.model
+            qset = model.objects.filter(datafile_id__in=file_ids).order_by("id")
+            count = qset.count()
+            total_deleted += count
+            logger.info(f"Deleting {count} records of type: {model}.")
+            handle_elastic_doc_delete(doc, qset, model, elastic_exceptions, new_indices)
+            qset._raw_delete(qset.db)
+        except DatabaseError as e:
+            log(
+                f"Encountered a DatabaseError while deleting records of type {model} from Postgres. The database "
+                "and Elastic are INCONSISTENT! Restore the DB from the backup as soon as possible!",
+                logger_context=log_context,
+                level="critical",
+            )
+            raise e
+        except Exception as e:
+            log(
+                f"Caught generic exception while deleting records of type {model}. The database and Elastic are "
+                "INCONSISTENT! Restore the DB from the backup as soon as possible!",
+                logger_context=log_context,
+                level="critical",
+            )
+            raise e
+    if elastic_exceptions != {}:
+        msg = (
+            "Warning: Elastic is inconsistent and the database is consistent. "
+            "Models which generated the Elastic exception(s) are below:\n"
+        )
+        for key, val in elastic_exceptions.items():
+            msg += f"Model: {key} generated {val} Elastic Exception(s) while being deleted.\n"
+        log(msg, logger_context=log_context, level="warn")
+    return total_deleted
+
+
+def delete_associated_models(meta_model, file_ids, new_indices, log_context):
+    """Delete all models associated to the selected datafiles."""
+    delete_summaries(file_ids, log_context)
+    delete_errors(file_ids, log_context)
+    num_deleted = delete_records(file_ids, new_indices, log_context)
+    meta_model.num_records_deleted = num_deleted
+
+
+def get_files_to_reparse(fiscal_year, fiscal_quarter, selected_files, reparse_all):
+    """Get the files to reparse."""
+    backup_file_name = "/tmp/reparsing_backup"
+    files = DataFile.objects.all()
+    continue_msg = "You have selected to reparse datafiles for FY {fy} and {q}. The reparsed files "
+    if selected_files:
+        files = files.filter(id__in=selected_files)
+        backup_file_name += "_selected_files"
```

Comment:
@raftmsohani I'm observing unexpected backup file names on selected files. See the 2 most recent backup file names below.

![Screenshot 2025-01-31 093748](https://github.com/user-attachments/assets/5fe8df7b-1ba3-4717-9bc9-90a7b3ef3b4d)